docs/02.API-REFERENCE.md

```diff
@@ -11924,3 +11924,76 @@ main (void)
   return 0;
 }
 ```
+
+# Container Functions
+
+## jerry_get_buffer_from_container
+
+**Summary**
+
+Return a new array containing elements from a Container, or a Container Iterator. Sets the boolean input value to `true`
+if the container object contains key-value structure and false if not.
+
+*Notes*
+- This API function depends on a build option (`JERRY_BUILTIN_CONTAINER`) and can be checked
+  runtime with the `JERRY_FEATURE_MAP, JERRY_FEATURE_SET, JERRY_FEATURE_WEAKMAP, JERRY_FEATURE_WEAKSET`
+  feature enum values.
+  see: [jerry_is_feature_enabled](#jerry_is_feature_enabled).
+- The es.next profile enables this by default.
+
+*New in version [[NEXT_RELEASE]]*.
+
+**Prototype**
+
+```c
+jerry_value_t
+jerry_get_buffer_from_container(jerry_value_t value,
+                                bool* is_key_value_p)
+```
+
+- `value` - Container object
+- `is_key_value` - Will be set to `true` if the given container has key-value pairs, `false` otherwise.
+- return value
+  - jerry_value_t containing an array of values of the underlying buffer, if the orignal is not empty
+  - jerry_value_t containing an empty array if the Container is empty, or the Iterator is finished
+  - Error if the `value` is nor a Container or a Container Iterator.
+  - `undefined` if the `value` is undefined/null.
+**Example**
+
+[doctest]: # ()
+
+```c
+#include "jerryscript.h"
+int
+main (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  jerry_char_t src[] = "var map = new Map(); map.set(1,2); map.entries()";
+  jerry_value_t iterable = jerry_eval (src, sizeof (src) - 1, JERRY_PARSE_NO_OPTS);
+
+  bool is_key_value_container = false;
+  jerry_value_t buffer_from_map = jerry_get_array_from_container (iterable, &is_key_value_container);
+  
+  if (jerry_get_array_length (buffer_from_map) == 2)
+  {
+    /* 
+      buffer_from_map contains will contain elements [1,2] as 1 is the key and 2 is the value from the
```

Comment:
The buffer_from_map contains two elements: 1 and 2, which is the key/value pair of the only item in the set.